src_c/polygon.c

```diff
@@ -315,6 +319,73 @@ pg_polygon_dealloc(pgPolygonObject *self)
     Py_TYPE(self)->tp_free((PyObject *)self);
 }
 
+static PyObject *
+pg_polygon_normal_polygon(PyObject *_null, PyObject *const *args,
+                          Py_ssize_t nargs)
+{
+    int sides;
+    double radius;
+    double angle = 0;
+    double Cx, Cy;
+
+    if (nargs == 3) {
+        if (!PyLong_Check(args[0])) {
+            goto error;
+        }
+        sides = (int)PyLong_AsLong(args[0]);
+
+        if (!pg_TwoDoublesFromObj(args[1], &Cx, &Cy) ||
+            !pg_DoubleFromObj(args[2], &radius)) {
+            goto error;
+        }
+    }
+    else if (nargs == 4) {
+        if (!PyLong_Check(args[0])) {
+            goto error;
+        }
+        sides = (int)PyLong_AsLong(args[0]);
+
+        if (!pg_TwoDoublesFromObj(args[1], &Cx, &Cy) ||
+            !pg_DoubleFromObj(args[2], &radius) ||
+            !pg_DoubleFromObj(args[3], &angle)) {
+            goto error;
+        }
+    }
```

Comment:
This code has a problem with `PyObject->int` conversion as `PyLong_AsLong` can fail, plus this code can be shaved off of some duplication:
```suggestion
  if (nargs < 3 || nargs > 4) {
      goto error;
  }
  if (!PyLong_Check(args[0])) {
        goto error;
  }
  sides = PyLong_AsLong(args[0]);
  if(PyErr_Occurred()) {
      return NULL;
  }
  if (!pg_TwoDoublesFromObj(args[1], &Cx, &Cy) ||
      !pg_DoubleFromObj(args[2], &radius)) {
      goto error;
  }
  if (nargs == 4 && !pg_DoubleFromObj(args[3], &angle) ) {
      goto error;
  }
```